imagepuller/main.go

```diff
@@ -34,17 +35,36 @@ func main() {
 		return
 	}
 
+	if err := newRootCmd().Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func newRootCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "imagepuller",
+		Short:        "pull and mount images",
+		Version:      version,
+		SilenceUsage: true,
+		RunE:         execute,
+	}
+	cmd.Flags().String("tmpdir", "", "temporary directory to use for storage")
+	return cmd
+}
+
+func execute(cmd *cobra.Command, _ []string) error {
```

Comment:
why have execute and run? you can use the use a single function. You can still set the flag manually on on cmd.Flags if you need to.